client/blocks/site-address-changer/index.jsx

```diff
@@ -214,11 +213,27 @@ export class SiteAddressChanger extends Component {
 		return currentDomainName.replace( currentDomainSuffix, '' );
 	}
 
+	/**
+	 * This is an edge case scenario where user have the site address changer opened and the user transfers
+	 * the site to atomic on other tab/window, losing sync between client and server. Client will try to
+	 * check availability against wordpress.com and will receive 404s because site is transfered to wpcomstaging.com
+	 */
+	isUnsyncedAtomicSite() {
+		const { validationError } = this.props;
+		const serverValidationErrorStatus = get( validationError, 'errorStatus' );
+
+		return serverValidationErrorStatus === 404;
+	}
+
 	getValidationMessage() {
 		const { isAvailable, validationError, translate } = this.props;
 		const { validationMessage } = this.state;
 		const serverValidationMessage = get( validationError, 'message' );
 
+		if ( this.isUnsyncedAtomicSite() ) {
+			return translate( 'wpcomstaging.com addresses cannot be changed.' );
```

Comment:
Is it definitely only Atomic sites where this happens? Should we make this more generic, to something like "This site's address cannot be changed"?